logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/PersistProxyCredentialsUseCase.kt

```diff
@@ -0,0 +1,32 @@
+package com.wire.kalium.logic.feature.client
+
+import com.wire.kalium.logic.StorageFailure
+import com.wire.kalium.logic.configuration.ProxyCredentialsRepository
+import com.wire.kalium.logic.functional.fold
+
+/**
+ * use case to persist the proxy credentials that been added from user while login, so it will be used
+ * to authenticate the proxy for the rest of the API calls in the app
+ */
+interface PersistProxyCredentialsUseCase {
+    suspend operator fun invoke(username: String, password: String): PersistProxyCredentialsResult
+
+    sealed class PersistProxyCredentialsResult {
+        object Success : PersistProxyCredentialsResult()
+        sealed class Failure : PersistProxyCredentialsResult() {
+            class Generic(val failure: StorageFailure) : Failure()
+        }
+    }
+}
+
+internal class PersistProxyCredentialsUseCaseImpl(
+    private val proxyCredentialsRepository: ProxyCredentialsRepository
+) : PersistProxyCredentialsUseCase {
+
+    override suspend operator fun invoke(username: String, password: String): PersistProxyCredentialsUseCase.PersistProxyCredentialsResult =
+        proxyCredentialsRepository.persistProxyCredentials(username, password).fold({
```

Comment:
tiny suggestion: since the repo name itself is explanatory why we need to have a full name function? isn't better to simply rename the `persistProxyCredentials` to `persist`?